route.go

```diff
@@ -89,34 +89,36 @@ func (r *Route) save() {
 	}
 }
 
-// Match check if the request match the route Pattern
-func (r *Route) Match(req *http.Request) bool {
+// Match check if the request matches the route Pattern and returns a map of the parsed variables
+// if it matches
+func (r *Route) Match(req *http.Request) (bool, map[string]string) {
 	ss := strings.Split(req.URL.EscapedPath(), "/")
-
 	if r.matchRawTokens(&ss) {
 		if len(ss) == r.Token.Size || r.Atts&WC != 0 {
-			vars.Lock()
-			if vars.v[req] == nil {
-				vars.v[req] = make(map[string]string)
+			totalSize := len(r.Pattern)
+			if r.Atts&REGEX != 0 {
+				totalSize += len(r.Compile)
 			}
+
+			vars := make(map[string]string, totalSize)
```

Comment:
Instead of letting the map dynamically re-allocate when we need it to.  I am pre-counting the space that will be needed and allocating exactly that amount of space.